File: pythonProject/composicao.py
class Motor:
    def __init__(self, potencia):
        self.potencia = potencia

class Carro:
    def __init__(self, marca, modelo, motor):
        self.marca = marca
        self.modelo = modelo
        self.motor = motor

    def exibir_detalhes(self):
        print(f'Marca: {self.marca}, Modelo: {self.modelo}, Motor: {self.motor.potencia} cavalos')

# Testando a composição
motor_v8 = Motor(450)
carro = Carro("Ford", "Mustang", motor_v8)
carro.exibir_detalhes()

File: GoLangProject/composicao.go
package main

import "fmt"

type Motor struct {
	Potencia int
}

type Carro struct {
	Marca  string
	Modelo string
	Motor  Motor
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Marca: %s, Modelo: %s, Motor: %d cavalos\n", c.Marca, c.Modelo, c.Motor.Potencia)
}

func main() {
	motorV8 := Motor{Potencia: 450}
	carro := Carro{Marca: "Ford", Modelo: "Mustang", Motor: motorV8}
	carro.ExibirDetalhes()
}
